pkg/mhooks/event.go

```diff
@@ -0,0 +1,308 @@
+package mhooks
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// ParseEvent returns a webhook event with the hydrated payload.
+//
+// Access the event payload by calling the corresponding getter method.
+// For example, if the event.EventType=account.created, call event.AccountCreated().
+//
+// The webhook signature is verified using the provided signing secret.
+func ParseEvent(r *http.Request, secret string) (*Event, error) {
+	isValid, err := checkSignature(r.Header, secret)
+	if err != nil {
+		return nil, fmt.Errorf("checking webhook signature: %w", err)
+	}
+
+	if !isValid {
+		return nil, errors.New("hashed signature using signing secret does not match value from x-signature header")
```

Comment:
I think this error should be exported so folks can match on it.

Reply:
good idea, updated here: https://github.com/moovfinancial/moov-go/pull/113/files#diff-1f5024b95de780def2e203c53d694c14068406cfab9a033d83a39aaa42e06119R11